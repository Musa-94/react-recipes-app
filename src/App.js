import React from 'react';
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';

function App() {
  return (
      <div> RECIPES </div>
  );
}

export default App;
