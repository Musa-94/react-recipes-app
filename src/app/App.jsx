import React from 'react';
import {
    Route,
    Switch,
    BrowserRouter as Router,
} from 'react-router-dom';
import './App.less';
import 'bootstrap/dist/css/bootstrap.min.css';

import Home from '../pages/Home.jsx';
import Recipes from '../pages/Recipes.jsx';
import SingleRecipe from '../pages/SingleRecipe.jsx';
import Default from '../pages/Default.jsx';
import Navbar from '../components/Navbar.jsx';

function App() {
  return (
      <Router>
          <main>
              {/* navbar */}
              <Navbar/>
              <Switch>
                  <Route path="/" exact component={Home} />
                  <Route path="/recipes" exact component={Recipes}/>
                  <Route path="/recipes/:id" component={SingleRecipe}/>
                  <Route component={Default}/>
              </Switch>
          </main>
      </Router>
  );
}

export default App;
