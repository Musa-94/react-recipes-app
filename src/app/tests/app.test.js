import React from 'react';
import App from '../App.jsx';

describe('App snapshot', () => {
    it('should correct renderer', () => {
        const component = renderer(<App/>);

        expect(component).toMatchSnapshot();
    });
});
