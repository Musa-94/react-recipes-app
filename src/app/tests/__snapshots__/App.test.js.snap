// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should correct renderer 1`] = `
<main>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="logo"
        src="test-file-stub"
      />
    </a>
    <div
      className="collapse navbar-collapse show ml-sm-5"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="navbar-item"
        >
          <a
            className="nav-link"
            href="#/"
          >
            Home
          </a>
        </li>
        <li
          className="navbar-item"
        >
          <a
            className="nav-link"
            href="#/recipes"
          >
            Recipes
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <header>
    <div
      className="container-fluid"
    >
      <div
        className="row align-items-center header-hero"
      >
        <div
          className="col text-center text-slanted"
        >
          <h1
            className="text-light text-uppercase display-3 letter-spacing text-slander"
          >
            amazing recipes
          </h1>
          <a
            className="text-uppercase btn btn-secondary btn-lg mt-3"
            href="#/recipes"
          >
            search recipes
          </a>
        </div>
      </div>
    </div>
  </header>
</main>
`;
